driver-core/src/test/unit/com/mongodb/internal/connection/TestConnectionPoolListener.java

```diff
@@ -28,52 +28,115 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 
 class TestConnectionPoolListener implements ConnectionPoolListener {
 
     private final List<Object> events = new ArrayList<Object>();
+    private final Lock lock = new ReentrantLock();
```

Comment:
Use a Lock + Condition to implement the `waitForEvent` operation

Reply:
Nice.